webv2/primarykey/primarykey.go

```diff
@@ -0,0 +1,244 @@
+				pkey.Desc = pkRequest.Columns[i].Desc
+				pkey.Order = pkRequest.Columns[i].Order
+
+				schemaissue := []internal.SchemaIssue{}
+
+				sessionState := session.GetSessionState()
+				schemaissue = sessionState.Conv.Issues[spannerTable.Name][pkey.Col]
+
+				if len(schemaissue) > 0 {
+
+					if contains(schemaissue, internal.Hotspot_AutoIncrement) {
+
+						schemaissue = Remove(schemaissue, internal.Hotspot_AutoIncrement)
+					}
+
+					if contains(schemaissue, internal.Hotspot_Timestamp) {
+
+						schemaissue = Remove(schemaissue, internal.Hotspot_Timestamp)
+					}
+
+					if contains(schemaissue, internal.Interleaved_Order) {
+
+						schemaissue = Remove(schemaissue, internal.Interleaved_Order)
+					}
+
+					if contains(schemaissue, internal.Interleaved_NotINOrder) {
+
+						schemaissue = Remove(schemaissue, internal.Interleaved_NotINOrder)
+					}
+
+					if contains(schemaissue, internal.Interleaved_ADDCOLUMN) {
+
+						schemaissue = Remove(schemaissue, internal.Interleaved_ADDCOLUMN)
+					}
+
+					schemaissue = []internal.SchemaIssue{}
+
+					sessionState.Conv.Issues[spannerTable.Name][pkey.Col] = schemaissue
+
+					if sessionState.Conv.Issues[spannerTable.Name][pkey.Col] == nil {
+
+						s := map[string][]internal.SchemaIssue{
+							pkey.Col: schemaissue,
+						}
+						sessionState.Conv.Issues = map[string]map[string][]internal.SchemaIssue{}
+
+						sessionState.Conv.Issues[spannerTable.Name] = s
+					} else {
+						sessionState.Conv.Issues[spannerTable.Name][pkey.Col] = schemaissue
+					}
+
+				}
+
+				list = append(list, pkey)
+			}
+		}
+	}
+	return list
+}
+
+// removePrimaryKey removed primary key from list of IndexKey.
+func removePrimaryKey(remove []int, spannerTable ddl.CreateTable) []ddl.IndexKey {
+
+	list := []ddl.IndexKey{}
+
+	for _, val := range remove {
+
+		colname := getColumnName(spannerTable, val)
+
+		for i := 0; i < len(spannerTable.Pks); i++ {
+
+			if spannerTable.Pks[i].Col == colname {
+
+				schemaissue := []internal.SchemaIssue{}
+				sessionState := session.GetSessionState()
+				schemaissue = sessionState.Conv.Issues[spannerTable.Name][spannerTable.Pks[i].Col]
+
+				if len(schemaissue) > 0 {
+
+					if contains(schemaissue, internal.Hotspot_AutoIncrement) {
+
+						schemaissue = Remove(schemaissue, internal.Hotspot_AutoIncrement)
+					}
+
+					if contains(schemaissue, internal.Hotspot_Timestamp) {
+
+						schemaissue = Remove(schemaissue, internal.Hotspot_Timestamp)
+					}
+
+					if contains(schemaissue, internal.Interleaved_Order) {
+
+						schemaissue = Remove(schemaissue, internal.Interleaved_Order)
+					}
+
+					if contains(schemaissue, internal.Interleaved_NotINOrder) {
+
+						schemaissue = Remove(schemaissue, internal.Interleaved_NotINOrder)
+
+					}
+
+					if contains(schemaissue, internal.Interleaved_ADDCOLUMN) {
+
+						schemaissue = Remove(schemaissue, internal.Interleaved_ADDCOLUMN)
+					}
```

Comment:
Better to use switch with fallthrough instead of multiple ifs

Reply:
Repetitive code , better to move it under a common method

Reply:
@shreyakhajanchi I have moved this into utils package as function   RemoveSchemaIssues()
https://github.com/searceinc/harbourbridge/blob/cc9c3ff1c64818c6cddf5ae3a06efb7840559c32/webv2/helpers/helpers.go#L113